drivers/audio/cxd56.c

```diff
@@ -3269,27 +3415,38 @@ static int cxd56_enqueuebuffer(FAR struct audio_lowerhalf_s *lower,
   FAR struct cxd56_dev_s *priv = (FAR struct cxd56_dev_s *)lower;
   struct audio_msg_s msg;
   irqstate_t flags;
+  int ret;
 
   flags = spin_lock_irqsave();
```

Comment:
@jerpelea 
I think this ```flag = spin_lock_irqsave();``` should be moved to inside the else condition.
Otherwise, a deadlock would happen if ```cxd56_src_enqueue()``` returns an error.